libecole/include/ecole/environment/environment.hpp

```diff
@@ -109,10 +116,18 @@ class Environment {
 			reward_function().before_reset(model());
 			information_function().before_reset(model());
 			auto const [done, action_set] = dynamics().reset_dynamics(model(), std::forward<Args>(args)...);
-
 			can_transition = !done;
+
+			OptionalObservation observation;
+			if (can_transition) {
+				observation = observation_function().extract(model(), done);
+			} else if constexpr (is_optional_v<Observation>) {
+				observation = Observation{};
```

Comment:
Can't you just return an OptionalObservation here ?

Reply:
Sorry, I'm not understanding you. What code would you like to see here?

Reply:
```suggestion
			if (can_transition) {
				observation = observation_function().extract(model(), done);
			} else {
				observation = OptionalObservation{};
			}
```

Reply:
You mean this?
```c++
if (!done) {
    observation = observation_function().extract(model(), done);
} else {
    observation = OptionalObservation{};
}
```

Reply:
In which case, I don't mind, if you think that's clearer. I think it's equivalent.

Reply:
Okay, changed that way.